### PR TITLE
OADP-612: Partially fail backup if VSB fails

### PR DESCRIPTION
**Pull request by eemcmullan**
Related: https://github.com/openshift/velero-plugin-for-csi/pull/40

### CODE REVIEW

File: pkg/controller/backup_controller.go [hunk 1]
```diff
@@ -655,9 +656,17 @@ func (c *backupController) runBackup(backup *pkgbackup.Request) error {
 
 		//DataMover case
 		if datamover.DataMoverCase() {
-			err = datamover.WaitForDataMoverBackupToComplete(backup.Name)
+			err = datamover.CheckIfVSBFailed(backup.Name)
+
+			// first check for failed VSB
```
Comment (shubham-pampattiwar) [inline pkg/controller/backup_controller.go]: Should we not just fail from plugin side ?

Comment (eemcmullan) [reply pkg/controller/backup_controller.go]: If we only fail from the plugin side then the wait for VSBs to complete still runs and times out

**Review by shubham-pampattiwar — APPROVED** [review]

File: pkg/datamover/datamover.go [hunk 2]
```diff
@@ -72,6 +77,14 @@ func CheckIfVolumeSnapshotBackupsAreComplete(ctx context.Context, volumesnapshot
 				if err != nil {
 					return false, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshotbackup %s/%s", volumesnapshotbackup.Namespace, volumesnapshotbackup.Name))
 				}
+
+				// check for a failed VSB
+				for _, cond := range tmpVSB.Status.Conditions {
+					if cond.Status == metav1.ConditionFalse && cond.Reason == ReconciledReasonError && cond.Type == ConditionReconciled {
```
Comment (kaovilai) [inline pkg/datamover/datamover.go:83]: Can we update this to return names of failed VSBs as well?

File: pkg/datamover/datamover.go [hunk 3]
```diff
@@ -111,7 +124,7 @@ func WaitForDataMoverBackupToComplete(backupName string) error {
 	if len(VSBList.Items) > 0 {
 		err = CheckIfVolumeSnapshotBackupsAreComplete(context.Background(), VSBList)
 		if err != nil {
-			log.Errorf("failed to wait for VolumeSnapshotBackups to be completed: %s", err.Error())
+			log.Errorf("volumeSnapshotBackup(s) failed to completed: %s", err.Error())
```
Comment (kaovilai) [inline pkg/datamover/datamover.go:127]: Ideally we can log which VSBs complete. Otherwise LGTM.

**Comment by eemcmullan** [comment]
closing for 1.9 branch

**Pull request closed without merging**